adr/minimal-sdk.md

```diff
@@ -0,0 +1,54 @@
+# ADR: Adopt a Minimal SDK and Ecosystem Strategy for Agents
+
+## Context
+
+Following the decision in [ADR: Adopt Llama Stack Responses API for Agentic MCP](https://github.com/opendatahub-io/agents/pull/1) to adopt the Llama Stack Responses API, we committed to providing a client-side "Agent construct" to manage state and orchestrate the agentic loop. This ADR addresses the critical follow-up question: what specific form will this construct take?
+
+The goal is to provide a solution that enables developers, offers a clear path for simple use cases, and integrates well with the broader AI ecosystem, all while managing Red Hat's engineering investment and long-term strategy.
+
+We considered several options for implementing this agent construct:
+
+1. **Extend Llama Stack Client SDK:** Add the agent logic directly into the official Llama Stack client. This would create a strong, unified story but conflicts with the client's design as a thin, auto-generated layer.
+2. **New, Full-Featured Red Hat SDK:** Build a comprehensive, competitive agent framework from the ground up. This offers maximum control but requires a very significant, long-term engineering investment. Also, even if it was done well, it would be difficult to drive adoption due to strong competition in this space.
+3. **New, Full-Featured Red Hat Container/Service:** The same as above, but delivered as a containerized HTTP service. This could simplify agent sharing and support a broader assortment of programming languages, but it adds operational complexity for developers.
+4. **Adopt a Single Third-Party SDK:** Formally adopt, package, and recommend a single existing framework (e.g., LangChain) as the "official" way to build agents on our platform. This would leverage a mature ecosystem but cedes strategic control and risks lock-in to that framework's architecture and release cadence.
```

Comment:
I think even _perceived_ lock-in to a single agent framework at this point would be a problem.

Reply:
One thing I want to ask is what what the scope of this ADR is. Is this just for the tech preview in 3.0, or more than that. For the scope of tech preview, this may be ok, but otherwise I would agree.